### PR TITLE
datastore: convert to minitest

### PR DESCRIPTION
**Pull request by quartzmo**
refs: #479

### CODE REVIEW

File: datastore/Gemfile [hunk 1]
```diff
@@ -15,11 +15,10 @@
 source "https://rubygems.org"
 
 gem "google-cloud-datastore"
-gem "sinatra"
```
Comment (quartzmo) [inline datastore/Gemfile:18]: @TheRoyalTnetennba Should `google-style` be required in this file? It is in #489 but not in #542.

File: datastore/Rakefile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "rake/testtask"
+
+Rake::TestTask.new "test" do |t|
+  t.test_files = FileList["**/*_test.rb"]
+  t.options = "--junit --junit-filename=sponge_log.xml"
+  t.warning = false
+end
```
Comment (quartzmo) [inline datastore/Rakefile:21]: @TheRoyalTnetennba Should there be an `acceptance ` task in this file? There is one in #489 but not in #542.

File: datastore/spec/quickstart_spec.rb [hunk 3]
```diff
@@ -1,55 +0,0 @@
...
-RSpec.configure do |config|
-  # show retry status in spec process
-  config.verbose_retry = true
-  # show exception that triggers a retry if verbose_retry is set to true
-  config.display_try_failure_messages = true
-
-  # set retry count and retry sleep interval to 10 seconds
-  config.default_retry_count = 5
-  config.default_sleep_interval = 10
-end
```
Comment (quartzmo) [inline datastore/spec/quickstart_spec.rb:28]: Does anyone think I should port this retry configuration to minitest? I did not notice it to be needed in several dozen local test runs, so I omitted it in hopes that it was added at a time when the service was having temporary issues.

File: datastore/Gemfile (hunk 1 above)
Comment (TheRoyalTnetennba) [reply datastore/Gemfile:18]: I kinda like it in the sample Gemfile b/c I run `rubocop -a` a lot to fix hash alignment etc.

File: datastore/spec/quickstart_spec.rb (hunk 3 above)
Comment (TheRoyalTnetennba) [reply datastore/spec/quickstart_spec.rb:28]: I have a bias towards retries in tests but idk if the client handles that already

File: datastore/Rakefile (hunk 2 above)
Comment (TheRoyalTnetennba) [reply datastore/Rakefile:21]: I don't think the acceptance task is needed atm. Wasn't originally sure how the ci for samples was going to look

File: datastore/Gemfile (hunk 1 above)
Comment (quartzmo) [reply datastore/Gemfile:18]: Ok I'll add it.

File: datastore/Rakefile (hunk 2 above)
Comment (quartzmo) [reply datastore/Rakefile:21]: Good, thanks!

File: datastore/spec/quickstart_spec.rb (hunk 3 above)
Comment (quartzmo) [reply datastore/spec/quickstart_spec.rb:28]: OK if we add it later when we actually experience service failures? I will be happy to.

File: datastore/Gemfile (hunk 1 above)
Comment (quartzmo) [reply datastore/Gemfile:18]: Added.

**Pull request merged**